Controller/ContentManagement/ContentTypeController.php

```diff
@@ -228,27 +228,15 @@ public function addAction(Request $request)
                     $environment = $contentTypeAdded->getEnvironment();
                     /** @var \Symfony\Component\HttpFoundation\File\UploadedFile $file */
                     $file = $request->files->get('form')['import'];
-                    $fileContent = file_get_contents($file->getRealPath());
-
-                    $encoders = array(new JsonEncoder());
-                    $normalizers = array(new JsonNormalizer());
-                    $serializer = new Serializer($normalizers, $encoders);
-                    /**@var ContentType $contentType */
-                    $contentType = $serializer->deserialize(
-                        $fileContent,
-                        "EMS\CoreBundle\Entity\ContentType",
-                        'json'
-                    );
+                    
+                    $contentType = $this->getContentTypeService()->readJson($file, $environment);
+                    
                     $contentType->setName($name);
                     $contentType->setSingularName($singularName);
                     $contentType->setPluralName($pluralName);
                     $contentType->setEnvironment($environment);
```

Comment:
Environment is allready passed to the readJson function, I think this call to setEnvironment can be removed!